analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -70,28 +77,30 @@ protected override void Initialize(SonarAnalysisContext context)
                             continue;
                         }
 
-                        if (typeSymbol.OriginalDefinition.IsAny(genericTypes))
+                        if (typeSymbol.OriginalDefinition.IsAny(GenericTypes))
                         {
                             return;
                         }
 
                         var suggestedGenericType = SuggestGenericCollectionType(typeSymbol);
                         if (suggestedGenericType != null)
                         {
-                            issues.Add(Diagnostic.Create(rule,
-                                        classDeclaration.Identifier.GetLocation(),
+                            issues ??= new();
+                            issues.Add(Diagnostic.Create(Rule,
+                                        typeDeclaration.Identifier.GetLocation(),
                                         suggestedGenericType));
                         }
                     }
 
-                    issues.ForEach(d => c.ReportIssue(d));
+                    issues?.ForEach(d => c.ReportIssue(d));
```

Comment:
Small syntax-sugar simplification
```suggestion
                    issues?.ForEach(c.ReportIssue);
```

Reply:
Why not? We're doing this quite often when possible

Reply:
Ah, indeed (the explanation was not visible before)